netlify/netlify-cms
# 1338 retain unknown fields

Closes #1338 where if a key was added to the frontmatter, but was excluded from the collection config, the CMS would not include the field in the editor form which meant it would be deleted upon the next save.

This PR enhances the selectField function by merging the fields configured in the collection with the fields that are present in the entry. When an unknown field is detected, it will be injected into the editor as a hidden field. This allows the data to be retained by using already established mechanisms and also prevents the UI from becoming cluttered.

**Test plan**

A test case has been added to the reducers/collections.spec.js suite. The test generates a collection with a single ```title``` field and an entry with a ```title``` field and an unknown field (to the collection) of ```tag```. After the field list is built, the test checks that both fields are present and that ```tag``` is using ```widget: 'hidden'```. If no entry is supplied, the selectField function does not have any change in behavior from previous versions.

Manual testing:

1. Add key/value to an entry's frontmatter, but not to the collection config
1. Edit the entry and save/publish normally
1. Check the frontmatter and the key/value pair should still exist

**A picture of a cute animal (not mandatory but encouraged)**

![Puppy fail](https://media.giphy.com/media/hgxbOOTbjSOOY/200w_d.gif)
